CONTRIBUTING.md

```diff
@@ -0,0 +1,172 @@
+- Triaging happens usually on Mondays or Thursdays
+- Review!
+  - Discuss the pr with one of the maintainers.
+  - Some changes might be requested.
+  - Review again.
+- Merge!
+
+## Development Environment Setup
+
+#### Requirements
+
+To install build requirements, run:
+```
+dnf builddep dnf5.spec #[--define '_without_<option> 1 ...]
+```
+
+#### Build
+
+To build DNF5, run:
+
+```
+mkdir build
+cd build
+cmake ..
+# or cmake .. [-DWITH_<OPTION>=<ON|OFF> ...]
+make
+```
+
+#### Test
+
+To run the tests, follow the steps to build the code and then run:
+
+```
+# from the 'build' directory
+CTEST_OUTPUT_ON_FAILURE=1 make test
+```
+
+As an alternative, tests can be executed in a verbose mode:
+
+```
+# from the 'build' directory
+make test ARGS='-V'
+```
+
+#### Install
+
+To install the built package, we recommend building an rpm package.
+
+To build rpms using tito, run:
+```
+tito build --rpm --test
+```
+
+To build rpms from git, run:
+
+```
+export PREFIX=$(rpmspec dnf5.spec -q --srpm --qf '%{name}-%{version}'); git archive --format=tar.gz --prefix=$PREFIX/ HEAD > $PREFIX.tar.gz
+rpmbuild -ba --define "_sourcedir $(pwd)" dnf5.spec #[--with=<option>|--without=<option> ...]
+```
+
+## Pull Request Checklist
+
+When you submit your pull request, or you push new commits to it, our automated
+systems will run some checks on your new code. We require that your pull request
+passes these checks, but we also have more criteria than just that before we can
+accept and merge it. We recommend that you check the following things locally
+before you submit your code:
+
+### Code should build
+
+Unless you need some help from us or you need us to review work in progress changes.
+Otherwise, please, [test it locally](#development-environment-setup).
+
+### Code should be tested
+
+If your pull request includes new or modified functionality within the library, we kindly request that you provide
+matching unit tests in the project's test directory to cover these changes. However, if the changes only affect
+the command-line interface, you may provide related CI tests in the
+[ci-dnf-stack](https://github.com/rpm-software-management/ci-dnf-stack) component.
+If you need assistance with this, please feel free to ask the maintainers for help.
+
+### Code should pass sanity checks
+
+Test the sanity of the codebase by performing [pre-commit](https://pre-commit.com/) checks.
+Inside `dnf5` directory run:
+```bash
+pre-commit install
+git add yourchanges
+git commit -m "your commit"
+...
+check the results, all light should be green
+
+```
+All these checks will run in as part of a PR action on GitHub, so be sure not to skip it if you don't want to
+amend your contribution.
+
+As part of pre-commit checks we perform checks such as trailing whitespaces, end of file fixes, clang-format and rpmlint
+checks. See [.pre-commit-config.yml](https://github.com/rpm-software-management/dnf5/blob/main/.pre-commit-config.yaml)
+for more info.
```

Comment:
Another thing occurred to me in this context. Maybe we could add a note about adding a proper description of the changes in the commit message and PR description?

Reply:
oh we could actually setup the commit template. what do you think?
https://github.com/rpm-software-management/dnf/blob/master/.git-commit-template

Reply:
But you still need to setup these template manually, right? Because I don't see the template by default using `git commit`. And also it seems not working properly in some IDEs like VS code from the link given.

If it couldn't be automated, I am rather for documenting the suggested format there.

Reply:
I created an issue to track this point. I believe there is a good margin of improvement and automation if we think this right.

https://github.com/rpm-software-management/dnf5/issues/498